hidimstat/estimator/_utils/u_Dnn_learner.py

```diff
@@ -0,0 +1,657 @@
+    ):
+        super().__init__()
+        if problem_type == "classification":
+            self.accuracy = Accuracy(task="multiclass", num_classes=output_dimension)
+        else:
+            self.accuracy = Accuracy(task="binary")
+        self.list_grps = list_grps
+        self.group_stacking = group_stacking
+        if group_stacking:
+            self.layers_stacking = nn.ModuleList(
+                [
+                    nn.Linear(
+                        in_features=len(grp),
+                        out_features=input_dim[grp_ind + 1] - input_dim[grp_ind],
+                    )
+                    # nn.Sequential(
+                    #     nn.Linear(
+                    #         in_features=len(grp),
+                    #         # out_features=max(1, int(0.1 * len(grp))),
+                    #         out_features=input_dim[grp_ind + 1]
+                    #         - input_dim[grp_ind],
+                    #     ),
+                    #     nn.ReLU(),
+                    # nn.Linear(
+                    #     in_features=max(1, int(0.1 * len(grp))),
+                    #     out_features=input_dim[grp_ind + 1]
+                    #     - input_dim[grp_ind],
+                    # ),
+                    # nn.ReLU(),
+                    # nn.Linear(
+                    #     in_features=max(1, int(0.1 * len(grp))),
+                    #     out_features=input_dim[grp_ind + 1]
+                    #     - input_dim[grp_ind],
+                    # ),
+                    # )
+                    for grp_ind, grp in enumerate(list_grps)
+                ]
+            )
+            input_dim = input_dim[-1]
+        self.layers = nn.Sequential(
+            # hidden layers
+            nn.Linear(input_dim, 50),
+            nn.ReLU(),
+            nn.Linear(50, 40),
+            nn.ReLU(),
+            nn.Linear(40, 30),
+            nn.ReLU(),
+            nn.Linear(30, 20),
+            nn.ReLU(),
+            # output layer
+            nn.Linear(20, output_dimension),
+        )
+        self.loss = 0
+
+    def forward(self, x):
+        if self.group_stacking:
+            list_stacking = [None] * len(self.layers_stacking)
+            for ind_layer, layer in enumerate(self.layers_stacking):
+                list_stacking[ind_layer] = layer(x[:, self.list_grps[ind_layer]])
+            x = torch.cat(list_stacking, dim=1)
+        return self.layers(x)
+
+    def training_step(self, batch, device, problem_type):
+        X, y = batch[0].to(device), batch[1].to(device)
+        y_pred = self(X)  # Generate predictions
+        if problem_type == "regression":
+            loss = F.mse_loss(y_pred, y)
+        elif problem_type == "classification":
+            loss = F.cross_entropy(y_pred, y)  # Calculate loss
+        else:
+            loss = F.binary_cross_entropy_with_logits(y_pred, y)
+        return loss
+
+    def validation_step(self, batch, device, problem_type):
```

Comment:
docstring ?

Reply:
This will be addressed by the issue #81.